File: applications/advection/2d/sphere/mapc2m.m

function [xp,yp,zp] = mapc2m(xc1,yc1)

global notpillowsphere;

% map = 'pillowsphere';
map = 'cubedsphere';

switch map
    case 'pillowsphere'
        notpillowsphere = false;
        [xp,yp,zp] = mapc2m_pillowsphere(xc1,yc1);
    case 'cubedsphere'
        [xp,yp,zp] = mapc2m_cubedsphere(xc1,yc1);
end


end

File: applications/advection/2d/sphere/mapc2m.f
      subroutine mapc2m(xc,yc,xp,yp,zp)
      implicit none

      double precision xc,yc,xc1,yc1,xp,yp,zp
      logical isdisk, iscart, ispillowsphere
      logical iscubedsphere

      xc1 = 2*xc - 1.d0
      yc1 = 2*yc - 1.d0

      if (iscart()) then
         call mapc2m_cart(xc1,yc1,xp,yp,zp)
      elseif (isdisk()) then
         call mapc2m_pillowdisk(xc1,yc1,xp,yp,zp)
      elseif (ispillowsphere()) then
         call mapc2m_pillowsphere(xc1,yc1,xp,yp,zp)
      elseif (iscubedsphere()) then
         call mapc2m_cubedsphere(xc,yc,xp,yp,zp)
      endif

      call scale_map(xp,yp,zp)
      call rotate_map(xp,yp,zp)

      end
